ui: style login page

couchand opened the pull request:
Adds some initial styling to the login page and the login indicator in the navigation sidebar.

Sidebar:
<img width="710" alt="screen shot 2018-05-29 at 1 59 44 pm" src="https://user-images.githubusercontent.com/793969/40676560-110e3a7e-6349-11e8-812a-9c8e682f5670.png">

Login page:
<img width="1274" alt="screen shot 2018-06-04 at 4 28 40 pm" src="https://user-images.githubusercontent.com/793969/40940739-c0c6eb10-6816-11e8-9b24-8a0b492239b3.png">

Login page with error:
<img width="1277" alt="screen shot 2018-06-04 at 4 45 40 pm" src="https://user-images.githubusercontent.com/793969/40940744-c4cb2d66-6816-11e8-90d4-d8b3c52927b6.png">

Insecure mode (temporary design):
<img width="161" alt="screen shot 2018-06-04 at 5 10 56 pm" src="https://user-images.githubusercontent.com/793969/40942388-016f9478-681c-11e8-929c-36d0b7aa7463.png">

Closes #24939.

couchand committed:
ui: declosurify event handlers on login page

Release note: None

couchand committed:
ui: style login page

Add some initial styling to the login page.

Contributes to #24939.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/26066)
<!-- Reviewable:end -->

couchand committed:
ui: error state for login page

Add some styling and clean up the error text on login page.

Contributes to #24939.

Release note: None

couchand committed:
ui: add cockroach icon to login page

Release note: None

couchand committed:
ui: style navbar login indicator

Adds some initial styling to the login indicator on the navigation sidebar.

Release note: None

couchand commented:
cc @josueeee for design review

couchand committed:
ui: add logout route

An explicit logout route can be useful, so this adds one.

Release note: None

vilterp commented:
Also, what does the sidebar look like in insecure mode?

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26066#-:-LDhbMtU2xTWjoX6CGxS:b9c5jnx)*
<!-- Sent from Reviewable.io -->

couchand commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[pkg/ui/src/views/login/loginPage.tsx, line 67 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/26066#-LDh5bCI1yT2HnAQpV67:-LEAfdVp7yc-qfXWapjR:b-iaj06y) ([raw file](https://github.com/cockroachdb/cockroach/blob/688608c0a763704502619ee3e0b61c814ec34b8d/pkg/ui/src/views/login/loginPage.tsx#L67)):*
<details><summary><i>Previously, vilterp (Pete Vilter) wrote…</i></summary><blockquote>

Can you add a comment noting why have to look at the response body (generated somewhere in the depths of `grpc-gateway` or `net/http` or somewhere), and not the status code?

I'm ok with doing this for now, but we should leave clues for our future selves to fix this.
</blockquote></details>

It's just here: https://github.com/cockroachdb/cockroach/blob/2137b24938fa2b888588b38e6915bd95952f3e4b/pkg/ui/src/util/api.ts#L160

But now that I've made a bunch of these hacks all over maybe it's time I actually fix it.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26066)*
<!-- Sent from Reviewable.io -->

couchand committed:
ui: fix ts errors from previous changes

Some TypeScript errors slipped through due to #26255.  This fixes them.

Release note: None

couchand committed:
ui, server: pass build tag to ui

This will get shown on the login page.

Release note: None

couchand committed:
ui: update styling of login page

This updates the login page styling to the latest designs.

Release note: None

couchand committed:
ui: style insecure mode indicator

This is temporary styling for the insecure mode indicator.  A final design will
be implemented later, but this will hold us over for now.

Release note: None

couchand committed:
ui: change copy on sign in button while waiting

Release note: None

vilterp commented on pkg/ui/src/views/app/components/layoutSidebar/index.tsx:
```diff
@@ -70,7 +71,13 @@ function LoginIndicator({ loginState, handleLogout }: LoginIndicatorProps) {
 
   if (!loginState.loginEnabled()) {
-    return (<div className="login-indicator">Insecure mode</div>);
+    return (
```
I don't see this indicator when I start a cluster in insecure mode. Just blank space above the cockroach icon.

couchand replied:
Are you sure you had the environment variable set?  I am seeing it...

vilterp commented on pkg/server/server.go:
```diff
@@ -1928,6 +1928,7 @@ func serveUIAssets(fileServer http.Handler, cfg Config) http.Handler {
 		tmplArgs := ui.IndexHTMLArgs{
 			ExperimentalUseLogin: cfg.EnableWebSessionAuthentication,
 			LoginEnabled:         cfg.RequireWebSession(),
+			Tag:                  build.GetInfo().Tag,
```
looks like this is what broke the `ServeIndexHTML` test in CI. It's pretty brittle — asserts the exact HTML that comes out. Not sure how to make that always pass, since the build tag will keep changing… Maybe can just sprintf it into the expected HTML.

couchand replied:
Tests updated.

vilterp commented on pkg/ui/src/routes/login.tsx:
```diff
@@ -1,12 +1,29 @@
 export const LOGIN_PAGE = "/login";
+export const LOGOUT_PAGE = "/logout";
+
+export default function(store: Store<AdminUIState>): JSX.Element {
```
nit: I don't like exporting anonymous functions; prefer to give them names (I think you can still export them in the same statement; maybe need an export statement below). Always giving functions names makes them easier to grep for and find via stack traces (as if we ever have reasonable stack traces… 😢 but that's a separate issue)

couchand replied:
Good point.  Done.

vilterp commented on pkg/ui/src/views/app/components/layoutSidebar/index.tsx:
```diff
@@ -78,11 +78,15 @@ function LoginIndicator({ loginState, handleLogout }: LoginIndicatorProps) {
+        >
+          { user[0] }
+        </div>
+        <Link to="/logout" onClick={handleLogout}>Sign Out</Link>
```
was gonna suggest that you make the whole thing clickable, but maybe the log out button should be pretty small so you don't click it accidentally.

couchand replied:
Yep.  Maybe one day we'll have a user menu of some kind?!

vilterp reviewed (approved):
Don't see insecure indicator in insecure mode. Otherwise looks good.

couchand committed:
ui: name route creators

A few files export functions used to configure routes.  These were
previously anonymous default exports, this change gives them names.

Release note: None

couchand commented:
bors r+

vilterp commented on pkg/ui/src/views/login/loginPage.tsx:
```diff
@@ -52,31 +63,74 @@ class LoginPage extends React.Component<LoginPageProps & WithRouterProps, LoginP
+              account access and password restoration.
+            </p>
+            <p className="aside">
+              <a href={docsURL("admin-ui-overview.html")} className="docs-link">
```
maybe change to https://www.cockroachlabs.com/docs/dev/admin-ui-access-and-navigate.html#secure-the-admin-ui in a followup?

vilterp replied:
@Amruta-Ranade will this link change?
